terraform/main.tf

```diff
@@ -102,3 +102,297 @@ resource "google_storage_bucket_iam_binding" "binding" {
+        value_source {
+          secret_key_ref {
+            secret = "superset-auth0-client-secret"
+            version = "2"
+          }
+        }
+      }
+      env {
+        name = "AUTH0_DOMAIN"
+        value_source {
+          secret_key_ref {
+            secret = "superset-auth0-domain"
+            version = "1"
+          }
+        }
+      }
+
+      ports {
+        container_port = 8088
+      }
+      resources {
+        limits = {
+          cpu    = "4"
+          memory = "2048Mi"
+        }
+      }
+    }
+    volumes {
+      name = "bucket"
+      gcs {
+        bucket    = google_storage_bucket.superset_storage.name
+        read_only = false
+      }
+    }
+    volumes {
+      name = "cloudsql"
+      cloud_sql_instance {
+        instances = ["catalyst-cooperative-pudl:us-central1:superset-database"]
+      }
+    }
+  }
+}
+
+resource "google_cloud_run_v2_service_iam_member" "noauth" {
+  location = google_cloud_run_v2_service.pudl-superset.location
+  name     = google_cloud_run_v2_service.pudl-superset.name
+  role     = "roles/run.invoker"
+  member   = "allUsers"
+}
+
+resource "google_secret_manager_secret" "superset_secret_key" {
+  secret_id = "superset-secret-key"
+  replication {
+    auto {}
+  }
+}
+
+resource "google_sql_database_instance" "postgres_pvp_instance_name" {
+  name             = "superset-database"
+  region           = "us-central1"
+  database_version = "POSTGRES_14"
+  settings {
+    tier = "db-custom-2-7680"
+    password_validation_policy {
+      min_length                  = 6
+      reuse_interval              = 2
+      complexity                  = "COMPLEXITY_DEFAULT"
+      disallow_username_substring = true
+      password_change_interval    = "30s"
+      enable_password_policy      = true
+    }
+  }
+  # set `deletion_protection` to true, will ensure that one cannot accidentally delete this instance by
+  # use of Terraform whereas `deletion_protection_enabled` flag protects this instance at the GCP level.
+  deletion_protection = true
+}
+
+resource "google_secret_manager_secret" "superset_database_username" {
+  secret_id = "superset-database-username"
+  replication {
+    auto {}
+  }
+}
+resource "google_secret_manager_secret" "superset_database_database" {
+  secret_id = "superset-database-database"
+  replication {
+    auto {}
+  }
+}
+resource "google_secret_manager_secret" "superset_database_password" {
+  secret_id = "superset-database-password"
+  replication {
+    auto {}
+  }
+}
+
+resource "google_secret_manager_secret" "superset_database_connection_name" {
+  secret_id = "superset-database-connection-name"
+  replication {
+    auto {}
+  }
+}
+
+resource "google_secret_manager_secret_iam_member" "superset_database_username_compute_iam" {
+  secret_id = google_secret_manager_secret.superset_database_username.id
+  role      = "roles/secretmanager.secretAccessor"
+  member    = "serviceAccount:345950277072-compute@developer.gserviceaccount.com"
```

Comment:
I think it makes sense to make a *new* service account that's only for Superset instead of using the default Compute Engine SA. Then it's easier to restrict access to the prod database, which makes it harder to accidentally blow it up.

Reply:
Ah yes it sure does. I'll create an issue for this.